proof/refine/ARM/Finalise_R.thy

```diff
@@ -3473,7 +3473,7 @@ crunches schedContextCompleteYieldTo, unbindNotification, unbindMaybeNotificatio
 lemma schedContextZeroRefillMax_unlive[wp]:
   "schedContextZeroRefillMax scPtr \<lbrace>\<lambda>s. P (ko_wp_at' (Not \<circ> live') p s)\<rbrace>"
   unfolding schedContextZeroRefillMax_def
-  apply (wpsimp wp: set_sc'.set_wp)
+  apply (wpsimp wp: set_sc'.set_wp simp: updateSchedContext_def)
```

Comment:
Not a big deal, but I'm a bit surprised to see `updateSchedContext` being unfolded all over the place.

Was this to minimise updates to existing proofs? If not, it might be a sign that we should prove more lemmas about it.

Reply:
For Finalise_R, it is mostly to save effort, as you say.

In Schedule_R and SchedContextInv_R, the proofs needed to be properly updated in most situations. Whether we should prove more lemmas or not is hard to say for sure, but I think it is ok to move forward and prove things on-demand as we proceed.